messenger.rst

```diff
@@ -1131,6 +1131,35 @@ during a request::
         :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\KernelTestCase`
         or :class:`Symfony\\Bundle\\FrameworkBundle\\Test\\WebTestCase`.
 
+Amazon SQS
+~~~~~~~~~~
+
+.. versionadded:: 5.1
+
+    The Amazon SQS transport has been added in Symfony 5.1
+    Install it by running:
+
+    .. code-block:: terminal
+
+        $ composer require symfony/amazon-sqs-messenger
+
+The ``SQS`` transport configuration looks like this:
+
+.. code-block:: bash
+
+    # .env
+    MESSENGER_TRANSPORT_DSN=sqs://guest:guest@sqs.eu-west-3.amazonaws.com/test?region=eu-west-3
+
+
+.. note::
+
+    By default, the transport will automatically create queue that are needed. That can be disabled.
+    
+The transport has a number of other options, including ways to configure
+the exchange, queues binding keys and more. See the documentation on
+:class:`Symfony\\Component\\Messenger\\Transport\\AmazonSqs\\Connection`.
+
```

Comment:
I would define the list of options:


Option | Description | Default
-- | -- | --
endpoint | Absolute URL to the SQS service | https://sqs.eu-west-1.amazonaws.com
region | Name of the AWS region | eu-west-1
queue_name | Name of the queue | messages
account | identifier of the AWS account | the owner of the credentials
access_key | AWS access key |
secret_key | AWS secret key
buffer_size | number of messages to prefetch | 9
wait_time | [long polling duration in seconds](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-short-and-long-polling.html) | 20
poll_timeout | wait for new message duration in seconds | 0.1
visibility_timeout | [amount of seconds the message won't be visible](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-visibility-timeout.html) | queue's configuration
auto_setup | Whether the table should be created automatically during send / get. | true

and add the note: 
```
difference between `wait_time` and `poll_timeout`: The `wait_time` parameter define 
the maximum duration Amazon SQS should wait until a message is available in a 
queue before sending a response. It helps reducing the cost of using Amazon SQS
by eliminating the number of empty responses.
The `poll_timeout` parameter define the duration the receiver should wait before 
returning null. It avoids blocking other receivers from being called 
```

and also add: 
```
If the queue name is suffixed by `.fifo` AWS will creates a 
[FIFO queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/FIFO-queues.html).
Use the stamp `Symfony\Component\Messenger\Bridge\AmazonSqs\Transport\AmazonSqsFifoStamp` 
to define the `Message group ID` and the `Message deduplication ID`.
```

Reply:
another note:
```
FIFO queues don't support setting a delay per message, for retries to work you need set this to 0 in the config.

retry_strategy:
                    delay: 0 # FIFO queues don't support per-message delay settings
```

I actually don't know if this should be in the docs or it is a bug :)